ksqldb-engine/src/test/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjectorTest.java

```diff
@@ -618,8 +653,7 @@ public void shouldThrowIfKeyFormatDoesNotSupportSchemaIdInference() {
 
     // Then:
     assertThat(e.getMessage(),
-        containsString("KEY_FORMAT should support schema inference when KEY_SCHEMA_ID is provided. "
-            + "Current format is KAFKA."));
+        containsString("KAFKA does not support the following configs: [KEY_SCHEMA_ID]"));
```

Comment:
I could not find where this change originates (ie, why does this error message change)?

Reply:
It's from:

https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjector.java#L241

and then

https://github.com/confluentinc/ksql/blob/5dd40134f6ad892cf092fd29ac2b0a3f60f37868/ksqldb-serde/src/main/java/io/confluent/ksql/serde/FormatFactory.java#L50

Since we put schema_id property to `FormatInfor` property and `kafka` format doesn't support this property, the validation will fail